src/Stan/Cli.hs

```diff
@@ -164,23 +166,52 @@ toggleSolutionP = flag ShowSolution HideSolution $ mconcat
     , help "Hide verbose solution information for observations"
     ]
 
+data ConfigCommand
+    = CheckCommand Check
+    | RemoveCommand Scope
+    | ObservationCommand (Id Observation)
+
+partitionCommands :: [ConfigCommand] -> ([Check], [Scope], [Id Observation])
+partitionCommands []                       = ([], [], [])
+partitionCommands (CheckCommand ch : rest) =
+    let (check, remove, obs) = partitionCommands rest
+    in (ch:check, remove, obs)
+partitionCommands (RemoveCommand r : rest) =
+    let (check, remove, obs) = partitionCommands rest
+    in (check, r:remove, obs)
+partitionCommands (ObservationCommand o : rest) =
+    let (check, remove, obs) = partitionCommands rest
+    in (check, remove, o:obs)
```

Comment:
I think we can remove a bit of code duplication here:

```haskell
partitionCommands (cmd : rest) =
    let (check, remove, obs) = partitionCommands rest
    in case cmd of ...
```